src/scheduler.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef __SCHEDULER__HPP_
+#define __SCHEDULER__HPP_
+
+#include <stddef.h>
+#include <cstdint>
+typedef uint32_t system_tick_t;
```

Comment:
This masks the Particle defined `system_tick_t` - this is not permitted in Particle